app/controllers/ui_job_wizard_controller.rb

```diff
@@ -23,15 +23,18 @@ def resource_name(nested_resource = nil)
     nested_resource || 'job_template'
   end
 
-  def map_template_inputs(template_inputs_with_foreign)
-    template_inputs_with_foreign.map { |input| input.attributes.merge({:resource_type => input.resource_type&.tableize }) }
-  end
-
   def resource_class
     JobTemplate
   end
 
   def action_permission
     :view_job_templates
   end
+
+  def resources
+    resource_type = params[:resource]
+    resource_list = resource_type.constantize.authorized("view_#{resource_type.underscore.pluralize}").all.map { |r| {:name => r.to_s, :id => r.id } }.select { |v| v[:name] =~ /#{params[:name]}/ }
+    render :json => { :results=>
+      resource_list.sort_by { |r| r[:name] }.slice(0..100), :subtotal => resource_list.count}
```

Comment:
Range is inclusive, so `0..100` means first 101 elements. Also `take` might be a better fit here

Reply:
didn't know about `take` thanks!